### PR TITLE
Reserve rerun perms for periodics

### PR DESCRIPTION
**Pull request by clarketm**
Periodics do not have an org, so using `"*"` allows this rerun config to apply to both **istio** and all the periodics.

### CODE REVIEW

File: prow/config.yaml
Comment (cjwagner) [inline prow/config.yaml:78]: Does this allow the istio/maintainers team to rerun istio-private jobs or is only the most specific rerun_auth_config respected?

**Review by cjwagner — APPROVED** [review]
I have one concern, but if it isn't substantive then this LGTM.
/hold

File: prow/config.yaml
Comment (clarketm) [reply prow/config.yaml:78]: It is only the *the most specific rerun_auth_config respected*.

**Pull request merged**
